kafka/publisher.go

```diff
@@ -0,0 +1,88 @@
+package kafka
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/dangkaka/go-kafka-avro"
+	queuesgo "github.com/merlinapp/queues-go"
+	"log"
+	"reflect"
+	"time"
+)
+
+type publisher struct {
+	producer   *kafka.AvroProducer
+	topic      string
+	schema     string
+	objectType reflect.Type
+}
+
+/*
+Creates a new Kafka publisher
+the objectType interface should be any of the following types, any other type will cause an error returning a nil value
+1. Copy of a structure
+2. Non-nil pointer to a struct of the expected type.
+If the structure doesn't have json tags, the schema will follow the literal fields names.
+*/
+func NewPublisher(kafkaServerAddress, schemaServerAddress, topic string, objectType interface{}) queuesgo.Publisher {
+	if !queuesgo.ValidateType(objectType) {
+		return nil
+	}
+	schema := createSchema(queuesgo.GetName(objectType), queuesgo.GetFields(objectType))
+	schemaBytes, err := json.Marshal(schema)
+	if err != nil {
+		return nil
+	}
+	fmt.Println("Schema registered: " + string(schemaBytes))
```

Comment:
fmt print necessary ?

Reply:
Yes! this is the schema registered on the schema registry server if the topic didn't exist
if that's not the case, then the schema should be exactly the same as the registered this helps a lot to verify